src/core/datastore/processors/cache-offline-data-processor.js

```diff
@@ -79,16 +79,23 @@ export class CacheOfflineDataProcessor extends OfflineDataProcessor {
   _processRead(collection, query, options) {
     let offlineEntities;
     return wrapInObservable((observer) => {
-      return this._ensureCountBeforeRead(collection, 'fetch the entities', query)
-        .then(() => super._processRead(collection, query, options))
+      return super._processRead(collection, query, options)
         .then((entities) => {
           offlineEntities = entities || []; // backwards compatibility
           observer.next(offlineEntities);
-          return this._networkRepository.read(collection, query, options);
+          return this._syncManager.pull(collection, query, options);
         })
-        .then((networkEntities) => {
-          observer.next(networkEntities);
-          return this._replaceOfflineEntities(collection, offlineEntities, networkEntities);
+        .then(() => {
+          return super._processRead(collection, query, options);
```

Comment:
Reading them again? I think you mentioned something about this yesterday, but could you elaborate?

Reply:
I am doing a read operation on the cache again because this was the only reliable way to get the full set of entities that match the original call after a delta set has been performed.